### PR TITLE
Error model Text Analytics

### PR DESCRIPTION
**Pull request by maririos**
This PR includes:
- Error model for Text analytics => #8864 
- Making the return types for single methods atomic
- Renaming of `RecognizeLinkedEntitiesResult.LinkedEntities` to `RecognizeLinkedEntitiesResult.LinkedEntities.Entities`

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/README.md [hunk 1]
```diff
@@ -147,8 +145,8 @@ Run a model to identify a collection of significant phrases found in the passed-
 ```C# Snippet:ExtractKeyPhrases
 string input = "My cat might need to see a veterinarian.";
 
-ExtractKeyPhrasesResult result = client.ExtractKeyPhrases(input);
-IReadOnlyCollection<string> keyPhrases = result.KeyPhrases;
+Response<IReadOnlyCollection<string>> response = client.ExtractKeyPhrases(input);
+IReadOnlyCollection<string> keyPhrases = response.Value;
```
Comment (annelo-msft) [inline sdk/textanalytics/Azure.AI.TextAnalytics/README.md]: Does this cast to IEnumerable?  If it does, the samples would be cleaner if we used that instead.

Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/README.md]: It does. Could you expand more on why?

Comment (annelo-msft) [reply sdk/textanalytics/Azure.AI.TextAnalytics/README.md]: IReadOnlyCollection is kind of esoteric, and developers will have to think about how to use it, what functionality it provides, etc.  IEnumerable is super common and most .NET developers can use it without thinking hard.  It's fine if it's an IReadOnlyCollection under the hood, but they don't need to know that.

**Review by willmtemple — APPROVED** [review]
all the names & shapes lgtm

File: sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsErrorTests.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+namespace Azure.AI.TextAnalytics.Tests
+{
+    public class TextAnalyticsErrorTests : SyncAsyncPolicyTestBase
+    {
+        public TextAnalyticsErrorTests(bool isAsync) : base(isAsync) { }
+
+        [Test]
+        public void DeserializeTextAnalyticsError()
+        {
+            var errors = @"
```
Comment (kristapratico) [inline sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsErrorTests.cs]: Could be worth adding a test where no innerError is returned. I think sending bad credentials will do this.

**Review by kristapratico — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeSentimentResult.cs [hunk 3]
```diff
@@ -14,17 +12,31 @@ namespace Azure.AI.TextAnalytics
...
         /// Gets the predicted sentiment for the full document.
         /// </summary>
-        public DocumentSentiment DocumentSentiment { get; }
+        public DocumentSentiment DocumentSentiment
+        {
+            get
+            {
+                if (HasError)
+                {
+#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
```
Comment (mssfang) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/AnalyzeSentimentResult.cs]: Just curious about if the line alignment is good?

**Review by mssfang — APPROVED** [review]
LGTM!

**Pull request merged**